tools/SourceKit/lib/SwiftLang/SwiftIndexing.cpp

```diff
@@ -134,10 +134,54 @@ class SKIndexDataConsumer : public IndexDataConsumer {
     if (!isRef) {
       uidAttrs = getDeclAttributeUIDs(symbol.decl);
       info.Attrs = uidAttrs;
+      if (auto *VD = dyn_cast<ValueDecl>(symbol.decl)) {
+        if (shouldOutputEffectiveAccessOfValueSymbol(symbol.symInfo)) {
+          AccessScope accessScope = VD->getFormalAccessScope();
+          UIdent AttrUID = SwiftLangSupport::getUIDForFormalAccessScope(accessScope);
+          info.EffectiveAccess = AttrUID;
+        }
+      }
     }
     return func(info);
   }
 
+  bool shouldOutputEffectiveAccessOfValueSymbol(SymbolInfo Info) {
+    SymbolKind Kind = Info.Kind;
+    SymbolSubKind SubKind = Info.SubKind;
+    switch (SubKind) {
+      case SymbolSubKind::AccessorGetter:
+      case SymbolSubKind::AccessorSetter:
+      case SymbolSubKind::SwiftAccessorWillSet:
+      case SymbolSubKind::SwiftAccessorDidSet:
+      case SymbolSubKind::SwiftAccessorAddressor:
+      case SymbolSubKind::SwiftAccessorMutableAddressor:
+      case SymbolSubKind::SwiftGenericTypeParam:
+        return false;
+      default:
+       break;
+    }
+    switch (Kind) {
+      case SymbolKind::Enum:
+      case SymbolKind::Struct:
+      case SymbolKind::Class:
+      case SymbolKind::Protocol:
+      case SymbolKind::Constructor:
+      case SymbolKind::EnumConstant:
+      case SymbolKind::Function:
+      case SymbolKind::StaticMethod:
+      case SymbolKind::Variable:
+      case SymbolKind::InstanceMethod:
+      case SymbolKind::ClassMethod:
+      case SymbolKind::InstanceProperty:
+      case SymbolKind::ClassProperty:
+      case SymbolKind::StaticProperty:
+      case SymbolKind::TypeAlias:
```

Comment:
Yes, this is nice, and I think a pretty complete list. One question: does `TypeAlias` include associated types? (If other protocol requirements have an effective access level, I think then so should associated types.)

Reply:
Yes, it's working the same as the properties 👍 
```
          key.kind: source.lang.swift.decl.associatedtype,
          key.name: "T",
          key.usr: "s:28index_effective_access_level26ProtocolWithAssociatedTypeP1TQa",
          key.line: 210,
          key.column: 20,
          key.effective_access: source.decl.effective_access.internal
```